src/topology/locally_constant/algebra.lean

```diff
@@ -110,4 +110,27 @@ instance [ring Y] : ring (locally_constant X Y) :=
 instance [comm_ring Y] : comm_ring (locally_constant X Y) :=
 { .. locally_constant.comm_semiring, .. locally_constant.ring }
 
+variables (M : Type*) [monoid M] (A : Type*) [semiring A]
+
+instance : has_scalar M (locally_constant X M) :=
+{ smul := λ a f,
+  { to_fun := λ x, a*f(x),
+    is_locally_constant := is_locally_constant.comp (is_locally_constant f)
+      (has_mul.mul a) } }
+
+instance : mul_action M (locally_constant X M) :=
+{ smul := (•),
+  one_smul := one_mul,
+  mul_smul := λ a b f, ext ( λ x, mul_assoc _ _ _ ) }
+
+instance : distrib_mul_action A (locally_constant X A) :=
+{ smul_add := λ r f g, ext ( λ x, mul_add _ _ _ ),
+  smul_zero := λ r, ext (λ x, mul_zero r),
+  ..locally_constant.mul_action A }
```

Comment:
Since the name for `locally_constant.mul_action` is auto-generated, I think this is more future-proof:
```suggestion
.. (by apply_instance : mul_action _ _) }
```

Reply:
Better would be
```lean
instance [monoid R] [add_monoid M] [distrib_mul_action R M] :
  distrib_mul_action R (locally_constant X M) :=
function.injective.distrib_mul_action coe_fn_add_monoid_hom coe_injective (λ _ _, rfl)
```
after defining
```
/-- `coe_fn` is an `add_monoid_hom`. -/
@[simps]
def coe_fn_add_monoid_hom [add_zero_class Y] : locally_constant X Y →+ (X → Y) :=
{ to_fun := coe_fn, map_zero' := rfl, map_add' := λ _ _, rfl }
```
further up the file.